File: innovacad_api/lib/src/domain/entities/course_module.dart
import 'package:vaden/vaden.dart';

@Component()
class CourseModule {
  String courses_modules_id;
  String course_id;
  String module_id;
  String? sequence_course_module_id;

  CourseModule({
    required this.courses_modules_id,
    required this.course_id,
    required this.module_id,
    this.sequence_course_module_id,
  });
}

File: innovacad_web/src/types/course_module.ts
type CourseModuleResponseData = {
  courses_modules_id: string | undefined;
  module_id: string | undefined;
  sequence_course_module_id: string | undefined;
  module_name: string | undefined;
  duration: number | undefined;
};

export type CourseModule = {
  courses_modules_id?: string;
  module_id: string;
  sequence_course_module_id?: string | null;
  module_name?: string;
  duration?: number;
};
